### PR TITLE
RN 0.50+ compat: main queue

### PR DESCRIPTION
**Pull request by kesha-antonov**

### CODE REVIEW

**Comment by airlaser** [comment]
+1 was about to open this myself. This is a simple, straightfoward fix that prevents the annoying yellow warnings now and avoids future bugs.

File: InAppUtils/InAppUtils.m [hunk 1]
```diff
@@ -19,6 +19,10 @@ - (instancetype)init
     return self;
 }
 
++ (BOOL)requiresMainQueueSetup {
+    return YES;
```
[the remark this answers is not in the document]
Comment (airlaser) [reply InAppUtils/InAppUtils.m]: From what I've seen a `NO` should be ok. I believe you only need a `YES` if you interact with the UI.

[the remark this answers is not in the document]
Comment (chirag04) [reply InAppUtils/InAppUtils.m]: Yup, that’s why I thought this should be no

**Comment by kesha-antonov** [comment]
@airlaser @chirag04 Set to NO 👍

**Comment by anonrig** [comment]
Can we merge this in?

**Comment by airlaser** [comment]
Can we make this 6.0.1?

**Comment by soheilpro** [comment]
Any idea when this fix will be released? The yellow warning is really annoying.

File: InAppUtils/InAppUtils.m [hunk 2]
```diff
@@ -19,6 +19,10 @@ - (instancetype)init
     return self;
 }
 
++ (BOOL)requiresMainQueueSetup {
+    return NO;
+}
```
Comment (oguchy) [inline InAppUtils/InAppUtils.m]: `+ requiresMainQueueSetup` は React Native 由来のメソッドでした。 `return NO` すると native module の初期化が非メインスレッドで行われうるようです
https://facebook.github.io/react-native/docs/native-modules-ios#implementing-requiresmainqueuesetup

**Comment by chirag04** [comment]
This is already released in 6.0.1

**Pull request merged**